packages/plugin-ext/src/common/plugin-ext-api-contribution.ts

```diff
@@ -19,33 +19,66 @@ import { interfaces } from '@theia/core/shared/inversify';
 
 export const ExtPluginApiProvider = 'extPluginApi';
 /**
- * Provider for extension API description
+ * Provider for extension API description.
  */
 export interface ExtPluginApiProvider {
     /**
-     * Provide API description
+     * Provide API description.
      */
     provideApi(): ExtPluginApi;
 }
 
 /**
- * Plugin API extension description.
- * This interface describes scripts for both plugin runtimes: frontend(WebWorker) and backend(NodeJs)
+ * Provider for backend extension API description.
```

Comment:
Hm. I must be confused. Historically, we've had just `ExtPluginApiProvider` which was used for both traditional plugins and web extensions. Now we have

- ExtPluginApiProvider
- ExtPluginBackendApiProvider
- ExtPluginFrontendApiProvider

It seems the third of these is unused and is maybe cruft from the early stages of implementing headless-plugin support. If that's right, then I also think `ExtPluginBackendApiProvider` should probably be named `ExtHeadlessPluginApiProvider`

Reply:
Nevermind. I think I might be confused. But maybe some additional documentation of these varying providers would help clarify the situation.